### ISSUE TITLE
[META] Improve test job runtimes

### ISSUE THREAD

**Issue by rpappalax** [mozilla-mobile/mobile-test-eng#2]
Test job runtimes (taskcluster query or script)

**Comment by rpappalax** [comment]
see:
[optimizations](https://docs.google.com/document/d/1-91YIZElvMisFdGbFq93g9bEI08jevJN9rmwcmg3BM8/edit#heading=h.4c9zqrvymorh)

**Comment by rpappalax** [comment]
@AaronMT do you feel that there's any more improvement that can be made here?

**Comment by AaronMT** [comment]
Yes, I think there's still room for improvement to explore different areas like test grouping but I don't think we need a meta bug.
